src/java.desktop/macosx/classes/sun/lwawt/macosx/CTrayIcon.java

```diff
@@ -215,13 +214,11 @@ void updateNativeImage(Image image) {
 
         CImage cimage = CImage.getCreator().createFromImage(image, observer);
         boolean imageAutoSize = target.isImageAutoSize();
-        if (cimage != null) {
-            cimage.execute(imagePtr -> {
-                execute(ptr -> {
-                    setNativeImage(ptr, imagePtr, imageAutoSize, useTemplateImages);
-                });
+        cimage.execute(imagePtr -> {
+            execute(ptr -> {
+                setNativeImage(ptr, imagePtr, imageAutoSize, useTemplateImages);
             });
-        }
+        });
```

Comment:
you deleted the wrong null check, the correct one is below see target == null below